marathon_acme/tests/test_sse_protocol.py

```diff
@@ -0,0 +1,285 @@
+        """
+        When the an unknown field is included in an event, it should be
+        ignored.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'data:hello\r\n')
+        self.protocol.dataReceived(b'somefield:123\r\n\r\n')
+
+        self.assertThat(data, Equals(['hello']))
+
+    def test_leading_colon_ignored(self):
+        """
+        When a line is received starting with a ':' character, the line should
+        be ignored.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'data:hello\r\n')
+        self.protocol.dataReceived(b':123abc\r\n\r\n')
+
+        self.assertThat(data, Equals(['hello']))
+
+    def test_missing_colon(self):
+        """
+        When a line is received that doesn't contain a ':' character, the whole
+        line should be treated as the field and the value should be an empty
+        string.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'data\r\n')
+        self.protocol.dataReceived(b'data:hello\r\n\r\n')
+
+        self.assertThat(data, Equals(['\nhello']))
+
+    def test_trim_only_last_newline(self):
+        """
+        When multiline data is received, only the final newline character
+        should be stripped before the data is passed to the callback.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'data:\r')
+        self.protocol.dataReceived(b'data:\n')
+        self.protocol.dataReceived(b'data:\r\n\r\n')
+
+        self.assertThat(data, Equals(['\n\n']))
+
+    def test_multiple_data_parts(self):
+        """
+        When data is received in multiple parts, the parts should be collected
+        to form the lines of the event.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(b'data:')
+        self.protocol.dataReceived(b' hello\r\n')
+        self.protocol.dataReceived(b'\r\n')
+
+        self.assertThat(data, Equals(['hello']))
+
+    def test_unicode_data(self):
+        """
+        When unicode data encoded as UTF-8 is received, the characters should
+        be decoded correctly.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        self.protocol.dataReceived(u'data:hëlló\r\n\r\n'.encode('utf-8'))
+
+        self.assertThat(data, Equals([u'hëlló']))
+
+    def test_line_too_long(self):
+        """
+        When a line is received that is beyond the maximum allowed length,
+        ``dataReceived`` should return a ``ConnectionLost`` error.
+        """
+        data = []
+        self.protocol.add_callback('message', data.append)
+        err = self.protocol.dataReceived(b'data:%s\r\n\r\n' % (b'x' * 16385,))
```

Comment:
Instead of hardcoding `16385` here, rather use `self.protocol.MAX_LENGTH` the build a long line or set `self.protocol.MAX_LENGTH` to a known (and much shorter) value.